website/docs/r/schema_site_contract_service_graph_listener.html.markdown

```diff
@@ -0,0 +1,178 @@
+    # 1. Administrative -> Security -> Certificate Authorities
+    # 2. Administrative -> Security -> Key Rings
+    target_dn = "uni/tn-example/certstore"
+  }
+}
+
+# Sample Listener configuration for the Network Load Balancer
+resource "mso_schema_site_contract_service_graph_listener" "example" {
+  contract_name = mso_schema_template_contract.example.contract_name
+  # Steps to configure Frontend IP Name
+  # 1. Application Management -> Services -> Create a "Network Load Balancer" -> Advanced Settings -> Additional Frontend IPs -> Frontend IP Names
+  frontend_ip_dn     = "uni/tn-example/clb-example/vip-2.2.2.2"
+  listener_name      = "example"
+  port               = 80
+  protocol           = "udp"
+  schema_id          = mso_schema.example.id
+  security_policy    = "default"
+  service_node_index = 0
+  site_id            = mso_schema_site_service_graph.example.site_id
+  template_name      = one(mso_schema.example.template).name
+  rules {
+    action_type       = "forward"
+    content_type      = "textPlain"
+    name              = "default"
+    port              = 80
+    priority          = 0
+    protocol          = "udp"
+    redirect_code     = "permMoved"
+    redirect_port     = 0
+    redirect_protocol = "inherit"
+    target_ip_type    = "unspecified"
+    url_type          = "original"
+    health_check {
+      interval            = 5
+      port                = 80
+      protocol            = "tcp"
+      success_code        = "200-399"
+      timeout             = 0
+      unhealthy_threshold = 2
+      use_host_from_rule  = "no"
+    }
+    provider_epg_ref {
+      anp_name      = mso_schema_template_anp.example.name
+      epg_name      = mso_schema_template_anp_epg_contract.example.epg_name
+      schema_id     = mso_schema.example.id
+      template_name = one(mso_schema.example.template).name
+    }
+  }
+}
+
+```
+
+## Argument Reference ##
+
+* `schema_id` - (Required) The schema ID of the Listener.
+* `template_name` - (Required) The template name of the Listener.
+* `contract_name` - (Required) The contract name of the Listener.
+* `site_id` - (Required) The site ID under which the Listener is deployed.
+* `service_node_index` - (Required) The service node index of the Service Graph under which the Listener is deployed.
+* `listener_name` - (Required) The name of the Listener.
+* `protocol` - (Required) The protocol of the Listener. Allowed values are `http`, `https`, `tcp`, `udp`, `tls` and `inherit`.
+* `port` - (Required) The port number of the Listener.
+* `security_policy` - (Optional) The security policy of the Listener. Allowed values are `default`, `eLBSecurityPolicy-2016-08`, `eLBSecurityPolicy-FS-2018-06`, `eLBSecurityPolicy-TLS-1-2-2017-01`, `eLBSecurityPolicy-TLS-1-2-Ext-2018-06`, `eLBSecurityPolicy-TLS-1-1-2017-01`, `eLBSecurityPolicy-2015-05`, `eLBSecurityPolicy-TLS-1-0-2015-04`, `AppGwSslPolicyDefault`, `AppGwSslPolicy20150501`, `AppGwSslPolicy20170401` and `AppGwSslPolicy20170401S`.
```

Comment:
For all the optional input with allowed values, is there default behaviour when not provided?

Reply:
The MSO/NDO setting it to `null` when the `security_policy` is not provided during the creation. That is not a valid option, So ignored the default value here.